### PR TITLE
Fix WC-Admin component styles for WP 5.3

### LINKED ISSUE #2978: Review WC-Admin styles in WP 5.3 Beta 1 environment
I recently updated my development environment to use WP 5.3 Beta 1 and while reviewing some wc-admin pulls today, I noticed that there seem to be some styling issues in the wc-admin with the new WP version. Examples are checkbox contracts seem off, alignment of checked inputs seems off, alignment of some select inputs seem off.

I think wp-admin made some styling tweaks to some admin inputs etc (sorry I don't know the trac tickets offhand) so it'd probably be a good idea to do a review of the wc-admin ui and document what needs fixed.

EDIT: Items to fix:

- [x] Leaderboard title font size
- [x] Bottom border leaderboard title
- [ ] Select field text alignment
- [ ] Input field borders?
~- [ ] Segmented control border~ _This didn't change_

### PR DESCRIPTION
**Pull request by jeffstieler**
Fixes #2978 

This PR seeks to apply the decisions made here: https://github.com/woocommerce/woocommerce-admin/issues/2978#issuecomment-558829197

>  `CompareFilter`: add border + border radius (from WP 5.3)
>  `DateRangeFilterPicker` & `DropdownButton`: add border + border radius, match chevron
>  `Search`: use purple focus from `SearchListControl`, add border + border radius
>  `SelectControl` & `TextControl` are both OBW-only (leave alone!)
>  `TextControlWithAffixes` - add border + border radius, no radius inboard of affixes
>  `StoreAlert` - use WP components for button & dropdown (form select)
> `Pagination` (and `StoreAlert` pagination): remove 3D look, vertical pipe between buttons

### Screenshots

(left is WP 5.2, right is 5.3)

Screen Shot 2019-12-05 at 11 22 39 AM
Screen Shot 2019-12-05 at 11 23 12 AM
Screen Shot 2019-12-05 at 11 23 33 AM
Screen Shot 2019-12-05 at 11 24 08 AM
Screen Shot 2019-12-05 at 11 24 50 AM
Screen Shot 2019-12-05 at 11 26 52 AM
Screen Shot 2019-12-05 at 11 26 16 AM

### Detailed test instructions:

This is where it gets time consuming..

- Spot check all WC-Admin pages (including DevDocs) on WP 5.2 and 5.3

### Changelog Note:

Fix: component styles for WordPress 5.3.

### CODE REVIEW

File: packages/components/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # 4.1.0 (Unreleased)
 - Added `<ImageUpload />` component.
+- Style form components for WordPress 5.3.
```
Comment (timmyc) [inline packages/components/CHANGELOG.md]: ⭐️

File: packages/components/src/text-control-with-affixes/index.js [hunk 2]
```diff
@@ -40,9 +41,15 @@ class TextControlWithAffixes extends Component {
 			describedby.push( `${ id }__suffix` );
 		}
 
+		const affixesClasses = classnames( 'text-control-with-affixes', {
+			'text-control-with-prefix': prefix,
+			'text-control-with-suffix': suffix,
+			disabled: this.props.disabled,
```
Comment (timmyc) [inline packages/components/src/text-control-with-affixes/index.js]: Looks like this is missing from the `propTypes` declaration.

Comment (jeffstieler) [reply packages/components/src/text-control-with-affixes/index.js]: Fixed in 022bc34.

**Comment by jeffstieler** [comment]
@LevinMedia thanks for reviewing - I'll take care of those issues in this PR.

**Comment by jeffstieler** [comment]
@LevinMedia fixed everything except:

> in devdocs the date range picker is mashed - looks fine on the site though

Can do in a follow up but it's not related to 5.3 (and isn't too big a deal).

**Comment by jeffstieler** [comment]
Note: CI builds will be fixed when https://github.com/woocommerce/woocommerce-admin/pull/3417 lands.

**Comment by jeffstieler** [comment]
Now that we've bumped the minimum WP version to 5.3, I've removed the version-specific styles. IF you have time to give this another once-over @LevinMedia, that would be great! 🙇

**Comment by LevinMedia** [comment]
Looks good!

**Pull request merged**